### PR TITLE
Allow EKS cluster creation when multiple AWS profile are defined

### LINKED ISSUE #5: Getting credentials: exec: executable aws failed with exit code 255
Running `terraform apply` in the cluster directory fails with this error message:
```
│ Error: Have got the following error while validating the existence of the ConfigMap "aws-auth": Get "https://xxx.gr7.eu-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps/aws-auth": getting credentials: exec: executable aws failed with exit code 255
│ 
│   with module.eks.kubernetes_config_map_v1_data.aws_auth[0],
│   on .terraform/modules/eks/main.tf line 553, in resource "kubernetes_config_map_v1_data" "aws_auth":
│  553: resource "kubernetes_config_map_v1_data" "aws_auth" {
```

Upon rerunning `terraform apply` it appears that it's failing to create this resource:
```
  # module.eks.kubernetes_config_map_v1_data.aws_auth[0] will be created
  + resource "kubernetes_config_map_v1_data" "aws_auth" {
      + data          = {
          + "mapAccounts" = jsonencode([])
          + "mapRoles"    = <<-EOT
                - "groups":
                  - "system:bootstrappers"
                  - "system:nodes"
                  "rolearn": "arn:aws:iam::xxx:role/dmz-eks-node-group-xxx"
                  "username": "system:node:{{EC2PrivateDNSName}}"
                - "groups":
                  - "system:bootstrappers"
                  - "system:nodes"
                  "rolearn": "arn:aws:iam::xxx:role/worker-eks-node-group-xxx"
                  "username": "system:node:{{EC2PrivateDNSName}}"
                - "groups":
                  - "system:masters"
                  "rolearn": "arn:aws:iam::xxx:role/connect-prod-radar-base-admin-role"
                  "username": "xxx-radar-base-admin-role"
            EOT
          + "mapUsers"    = jsonencode([])
        }
      + field_manager = "Terraform"
      + force         = true
      + id            = (known after apply)

      + metadata {
          + name      = "aws-auth"
          + namespace = "kube-system"
        }
    }
```

### PR DESCRIPTION
**Pull request by keyvaann**
Aiming to fix #5. 
Also added the cluster prefix to the node groups so they'd be easier to distinguish EC2 servers list. **Note:**  This will delete and recreate node groups, might cause slight disruptions.

### CODE REVIEW

**Comment by baixiac** [comment]
Another finding is the limit on the length of the name_prefix for a node group. If the cluster name is set to "dev-radar-base-cluster" which used to work,  the following errors will occur:
```
│ Error: expected length of name_prefix to be in the range (1 - 38), got dev-radar-base-cluster-dmz-eks-node-group-
│ 
│   with module.eks.module.eks_managed_node_group["dmz"].aws_iam_role.this[0],
│   on .terraform/modules/eks/modules/eks-managed-node-group/main.tf line 421, in resource "aws_iam_role" "this":
│  421:   name_prefix = var.iam_role_use_name_prefix ? "${local.iam_role_name}-" : null
│ 
│ Error: expected length of name_prefix to be in the range (1 - 38), got dev-radar-base-cluster-worker-eks-node-group-
│ 
│   with module.eks.module.eks_managed_node_group["worker"].aws_iam_role.this[0],
│   on .terraform/modules/eks/modules/eks-managed-node-group/main.tf line 421, in resource "aws_iam_role" "this":
│  421:   name_prefix = var.iam_role_use_name_prefix ? "${local.iam_role_name}-" : null
```

File: cluster/provider.tf [hunk 1]
```diff
@@ -11,7 +12,7 @@ provider "kubernetes" {
 
   exec {
     api_version = "client.authentication.k8s.io/v1beta1"
-    args        = ["eks", "get-token", "--cluster-name", module.eks.cluster_name, "--region", var.AWS_REGION]
+    args        = ["eks", "get-token", "--cluster-name", module.eks.cluster_name, "--region", var.AWS_REGION, "--profile", var.AWS_PROFILE]
```
[the remark this answers is not in the document]
Comment (K1Hyve) [reply cluster/provider.tf:15]: I'm not sure what the best practice to handle terraform variables but I'm using tfvar files at the moment to store the configuration which also includes AWS variables, I think as long as it's possible to specify a profile via tfvar files the rest shouldn't be an issue and we can implement it differently.

**Comment by K1Hyve** [comment]
I don't know where `eks-node-group` part is coming from since I couldn't find it in the code but if we change it to `eks-ng` it should cover most names.

**Comment by baixiac** [comment]
That part must have come from the eks module and longer names could violate the AWS rule on naming the Node IAM role. I can see the "Security group name" column does contain the cluster name although that's probably not ideal.

**Comment by keyvaann** [comment]
Fixed the conflicts and things seems to be working now.
Had to make access key and secret key optional to make Terraform read the AWS profile.

**Comment by keyvaann** [comment]
@baixiac this PR is also ready for review

**Review by baixiac — APPROVED** [review]
LGTM

**Pull request merged**